### ISSUE TITLE
[Feature Request] Add commands for "Go/Save to last used folder/location"

### ISSUE THREAD

**Issue by tomas-chrastina** [alainfrisch/nostalgy#70]
Hello!

From time to time there are situations when I really miss easy keyboard shortcut / command for:
- **Go to last used folder**
- **Save to last used folder**

So I thought it would be beneficial to add them.

Thank you in advance for considering this.

--
Best regards,
Tomas Chrastina.

**Comment by Markmull** [comment]
I updated Thunderbird somewhat recently to 68.8.0 & now Nostalgy is not as keyboard friendly. For example, after "s" for saving to a location, the recent locations still pop up, but it does not highlight one or allow me to use the up arrow to highlight one & hit Enter to move it. I must first begin typing part of the location I want, only then can I use the up arrow to highlight the location to move it there. There are some other functionality losses similar to that. I may look at backing out of recent updates. Nostalgy is basically a killer-addon for me. It's become _essential_.

**Comment by opto** [comment]
yes, that is still written, but into a box which is hidden by Thunderbird design. Trying to make it visible for TB 68 but still failing.

**Comment by JoelAzaria** [comment]
@tomas-chrastina @opto 
That feature is already baked into Nostalgy (pre TB68) with the shift key modifier.

So eg. shift+s will save to the last-used folder and shift+g will go there.

However since TB68 broke the extension and @opto has been making alterations to get it working, I think shift+ hasn't been looked at yet.

@opto, I noticed this shift+ behavior mentioned in another thread (as an unanswered question) so I thought it appropriate to clarify here.  hth.

Thanks for taking up the reigns btw.  Nostalgy is important to many.  Will be adding to the BountySource soon as well.  Hope you can get this thing back to it's former glorious state  :)

**Comment by opto** [comment]
shift + is updated with respect to working logic. But statusbar has been dropped/changed, and the box with the choices is not shown (yet). So it cannot be selected. Needs some more digging into current TB user interface. For statusbar, there are many nested layers, although I set the box visible  (and verified in the debugger), it is then reset to unvisible in a place I did not find yet.

**Comment by tomas-chrastina** [comment]
@JoelAzaria: I'm using version 60.9.0. But maybe I missed something.
My scenario is like this:
1) I'm in folder **Inbox**.
2) I move message **Message1** to suggested folder using **SHIFT+S** to folder **Folder1**.
3) Focus in Inbox will move to next message - **Message2**, that is suggesting folder **Folder2**.
4) Now I remembered I'd like to go to folder I used last save operation. Or just after a while when I did work outside of Thunderbird.
It means go to last used folder = **Folder1**, NOT currently suggested **Folder2**.
And this is what I would like to have shortcut for.

**Comment by eliotmoss** [comment]
On 10/28/2019 8:29 AM, Tomas Chrastina wrote:
> @JoelAzaria <https://github.com/JoelAzaria>: I'm using version 60.9.0. But maybe I missed something.
> My scenario is like this:
> 
>  1. I'm in folder *Inbox*.
>  2. I move message *Message1* to suggested folder using *SHIFT+S* to folder *Folder1*.
>  3. Focus in Inbox will move to next message - *Message2*, that is suggesting folder *Folder2*.
>  4. Now I remembered I'd like to go to folder I used last save operation. Or just after a while when
>     I did work outside of Thunderbird.
>     It means go to last used folder = *Folder1*, NOT currently suggested *Folder2*.
>     And this is what I would like to have shortcut for.

In the older version of the world I am working in, _rules_ can be used to suggest
a default folder.  shift+S will save there.  To save into a different, but recent
folders, one types s, and then uses up/down arrows to select the desired recent
folder, then Enter to save there.  Thus, to save into the most recent folder,
s, up, Enter.

I would _not_ want shift+S to take away my ability to use the suggested rule.

Best wishes - EM

**Comment by tomas-chrastina** [comment]
@eliotmoss: I don't want to take away anything. SHIFT + S is just fine.
I would like to enhance it a bit. So according to my mentioned scenario, it would be nice to have configurable keyboard shortcut to go directly to folder *Folder 1* even when current message suggest to use different one. To basically save last used folder to "lastUsedFolderVariable". And then even when I will move whenever among folders, changing focus on different messages etc., when I press let's say CTRL + SHIFT + G (could be anything else in configuration) I will be moved/navigated to *Folder 1* (or any last used folder for save/go to operation).

**Comment by opto** [comment]
Uploaded, so give it a try. Use on test-TB only, not on your most valuable emails.
https://github.com/opto/nostalgy-xpi,

**Comment by caugner** [comment]
@tomas-chrastina Just to make sure I correctly understand your feature request:

* Nostalgy should record in which folders you have *been* last (without necessarily having executed an action there).
* Nostalgy should provide a key shortcut to toggle between the last visited folder ("toggle", because pressing the shortcut twice would return to the same folder).

Is that your intention?

**Comment by tomas-chrastina** [comment]
@caugner Probably. If as co called LastFolder we mean last folder used by Nostalgy, not folder I focused on in Thunderbird by navigation in folder three, ....

So when I use Nostalgy shortcut for Save/GoTo folder, store this folder to e.g. `extensions.nostalgy.last_used_folder`.

Then by using "new" shortcut, use automaticaly this folder with operation GoTo/Save. Something similar to last(I guess it is the last one in the list) folder in `extensions.nostalgy.recent_folders`. But I guess there is a need to have one folder per server when we set `extensions.nostalgy.restrict_to_current_server` to `true`? Just thinking out loud, I am not sure how exactly this option is handled in context to recent folders.

**Comment by JoelAzaria** [comment]
As this feature is already built in (and I think Opto said above he fixed it after it broke for 68) I think possibly and easy/cheap and maybe better solution is to have a separate (user definable?) key combo for the "suggested" (as per the rules) folder and leave shift+s for the last folder used by Nostalgy.
In this way it's not an either/or for suggested vs last used but both can be used at once.
For those that like the 'old' (ie present) behavior, lets say that setting the "suggested" to the same keystroke - shift+s - overrides and then supersedes the "last folder" behavior.
imho that should work, serve all angles talked about here and (my guess is) be relatively easy to implement in terms of coding.

Thoughts?
